quickstart/dotnet-core/Program.cs

```diff
@@ -207,102 +208,97 @@ public static async Task ForceReconnectAsync()
 
         // In real applications, consider using a framework such as
         // Polly to make it easier to customize the retry approach.
-        private static async Task<T> BasicRetryAsync<T>(Func<T> func)
+        private static async Task<T> BasicRetryAsync<T>(Func<IDatabase, Task<T>> func)
         {
             int reconnectRetry = 0;
-            int disposedRetry = 0;
 
+            IDatabase cache = Connection.GetDatabase();
             while (true)
```

Comment:
Can we store the database in a class field rather than call .GetDatabase() in every call to BasicyRetryAsync()? 
We should also name it `database` rather than `cache`